command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -62,6 +80,30 @@ func (c *cmd) Run(args []string) int {
 		return 1
 	}
 
+	if c.enhance {
+		if _, err := f.Seek(0, 0); err != nil {
+			c.UI.Error(fmt.Sprintf("Error resetting file for enhancement, got %s", err))
+		}
+		decomp, err := gzip.NewReader(f)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("failed to decompress snapshot: %v", err))
+			return 1
+		}
+		defer decomp.Close()
+
```

Comment:
After you decompress the `f` reader into `decomp` you have to pass it through an un-tar operation and extract the `state.bin` file from within the tar envelope, then _that_ is what goes to `enhance`